### DISCUSSION TITLE
Doric File Compatibility

### DISCUSSION THREAD

**Discussion by Obarnhill** [bahanonu/calciumImagingAnalysis#134] [Q&A] [answered]
Hi all. Does anyone know if CIAtah is compatible with doric files? I cannot visualize my recordings using CIAtah even when I change the file extension to .h5. I know the Doric files are supposed to be HDF5 files, so I am not sure what the issue is here (they are HDF5 with a .doric extension). I tried also visualizing the examples from the downloaded CIAtah folder, and I could not visualize them in Doric's software but could work with them well in CIAtah. If anyone knows how to make these compatible, I would love to hear it. Thansk! 

Best,
Olivia

**Comment by bahanonu** [comment]
Hi Olivia, @Obarnhill 

Could you provide a download link for one of the doric files? If they are true HDF5 files then they should be compatible with HDF5, I'll just need to check what the dataset name/path is and then you can update that in the CIAtah GUI as below (see red arrow). That would also likely help making them compatible with Doric software, assuming their files do not require additional metadata.

![image](https://github.com/bahanonu/ciatah/assets/5241605/fabd2237-5603-4771-910d-e6153f0fd4c4)

cheers,
-Biafra

**Comment by Obarnhill** [comment]
I uploaded a file to google drive. Here is the link. Thank you! I can also send a picture of the results from using h5disp on this file if you would prefer.
https://drive.google.com/file/d/1v0gEbOGrL99DvpeaNDd3oUk7TIrprtqO/view?usp=sharing

**Comment by bahanonu** [comment, marked as answer]
Thanks!  Looks like Doric is storing the data in `/DataAcquisition/FMD/Images/Series0001/Sensor1/ImagesStack` but their format is `T X Y` compared to `X Y T` that I use. 

Can load and convert the movies to `X Y T` for now using the below, I'll see about adding Doric conversion module (to allow batch converting or depending on your file format I can send a script) or direct support (e.g. my loading functions will just reshape) so you can avoid having to re-save the data.

```MATLAB
%% Load .doric movie
inputFilePath = 'FULLPATH\concat_obMiniscope_230922_Habituation_0000.h5';
inputDatasetName = '/DataAcquisition/FMD/Images/Series0001/Sensor1/ImagesStack';
inputMovie = ciapkg.io.loadMovieList(inputFilePath,'inputDatasetName',inputDatasetName);

% Doric files in [T X Y], switch to [X Y T]
inputMovie2 = permute(inputMovie,[2 3 1]);

disp('Done!')
%% View movie (comment out when batching)
ciapkg.view.playMovie(inputMovie2);

%% Save movie out in [X Y T]
outputFilePath = 'FULLPATH\concat_obMiniscope_230922_Habituation_0000.h5';
ciapkg.io.saveMatrixToFile(inputMovie2,outputFilePath,'inputDatasetName',inputDatasetName);
disp('Done!')
```

If I process the data you sent with CIAtah, I get PCA-ICA outputs as below so looks to be working fine. (As an aside, where are you imaging?)

![image](https://github.com/bahanonu/ciatah/assets/5241605/3663e448-8281-4b66-b785-d69db87aceb6)


As a heads up, if you select `Preview file internal structure` in the main CIAtah GUI, it will produce an abbreviated version of h5disp.

![image](https://github.com/bahanonu/ciatah/assets/5241605/89e51b69-e7f3-4470-b1fb-7ece0d1be7ff)

**Comment by Obarnhill** [comment]
Thank you!!! Works great. This is my first time with HDF5 files, so the structure didn't make much sense until now. Thanks for clearing that up.

I am in dorsolateral striatum of a2a-cre mouse and visualizing cre-dependent GCaMPP6s. This is just a pilot mouse to set up miniscope in the lab. 

Olivia

**Comment by bahanonu** [comment]
Great! Let me know if you have any other questions.

**Reply by Obarnhill** [reply]
Hi Biafra, I was able to get through pre-processing but am having trouble with PCA-ICA cell extraction. I get an error every time I try to run it. Weirdly CNMFE runs fine (although doesn't label many cells). What parameters did you use to achieve the image above? Or do you have any other suggestions as to why I am getting an error? Thanks again for your help.

**Reply by bahanonu** [reply]
Glad to hear CNMF-e runs fine, are # of cells similar to what I posted and re: PCA-ICA, can you paste a screenshot or code snippet of the errors?

For that image, I used mu=0.1, term = 5e-06, and max iter = 1000 (then did manual sorting after).

**Reply by Obarnhill** [reply]
I played around with it more and am now getting similar results. Thanks as always for your help! 

Olivia
